internal/user/api.go

```diff
@@ -0,0 +1,18 @@
+package user
+
+import "github.com/tilt-dev/tilt/pkg/model"
+
+// TODO(nick): Eventually would like this interface to help with
+// 1) other kinds of user preferences (token? analytics opt-in/opt-out?)
+// 2) server-based preferences
+type Prefs struct {
+	// The kind of metrics stack the user is talking to.
+	MetricsMode model.MetricsMode `json:"metricsMode,omitempty" yaml:"metricsMode,omitempty"`
+}
+
+// Read/write metrics setting from a store.
+// Inspired loosely by https://pkg.go.dev/k8s.io/client-go/kubernetes/typed/core/v1#PodInterface
+type PrefsInterface interface {
```

Comment:
Just a naming nit, FFTI: isn't naming an interface `PrefsInterface` redundant? I'd just do `Prefs` and call the struct `prefs`, or call the interface `Preferences`/the struct `Prefs`, or something.

Reply:
See the comment 1 line above this, this is taking the Kubernetes naming convention of [ResourceName]Interface.

I'm not a big fan of the convention, but think the benefits of copying another convention outweighs the costs

Reply:
womp womp. Cool!